annotation_converters/objectron_2_coco.py

```diff
@@ -0,0 +1,198 @@
+import argparse
+import json
+import os
+from os import path as osp
+
+from tqdm import tqdm
+import cv2 as cv
+import numpy as np
+
+from objectron_helpers import load_annotation_sequence, grab_frames, get_video_frames_number
+
+lists_root_path = osp.abspath(os.path.join(osp.dirname(__file__), '../3rdparty/Objectron/index'))
+
+
+def load_video_info(data_root, subset, classes):
+    videos_info = []
+    avg_vid_len = 0
+    for cl in classes:
+        with open(osp.join(lists_root_path, f'{cl}_annotations_{subset}'), 'r') as f:
+            for line in f:
+                ann_path = osp.join(data_root, 'annotation' + osp.sep + line.strip() + '.pbdata')
+                ann = load_annotation_sequence(ann_path) # object class can be incorrect in annoatation
+                for item in ann:
+                    item[1] = cl
+                assert len(ann) > 0
+                avg_vid_len += len(ann)
+                vid_path = osp.join(data_root, 'videos' + osp.sep + line.strip() + osp.sep + 'video.MOV')
+                videos_info.append((vid_path, ann))
+
+    avg_vid_len /= len(videos_info)
+    return videos_info, avg_vid_len
+
+
+def np_encoder(object):
+    if isinstance(object, np.generic):
+        return object.item()
+
+
+def decode_keypoints(keypoints, keypoint_size_list, size):
+    keypoints = np.split(keypoints, np.array(np.cumsum(keypoint_size_list)))
+    keypoints = [points.reshape(-1, 3) for points in keypoints]
+    unwrap_mat = np.asarray([size[0], size[1], 1.], np.float32)
+    keypoints = [
+        np.multiply(keypoint, unwrap_mat).astype(int)[:, :-1]
+            for keypoint in keypoints
+    ][:len(keypoint_size_list)]
+    return keypoints
+
+
+def get_bboxes_from_keypoints(keypoints, num_objects, size):
+    w, h = size
+    bboxes = []
+    num_valid = 0
+
+    for i in range(num_objects):
+        min_x = np.min(keypoints[i][:,0])
+        min_y = np.min(keypoints[i][:,1])
+        max_x = np.max(keypoints[i][:,0])
+        max_y = np.max(keypoints[i][:,1])
+        bbox = [min_x, min_y, max_x - min_x, max_y - min_y]
+        if min_x < 0 or min_y < 0 or max_x >= w or max_y >= h and bbox[2]*bbox[3] == 0:
```

Comment:
maybe do not delete all of the bbox when they mismatch image resolution, but clamp coordinates first and then ensure that the length of the sides of the boxes does not shrink to zero.

Reply:
If we do so without any threshold, a lot of hard outliers will arise. I do not observe critical for the detection reduction of data, so prefer to keep the annotation cleaner.